src/detection_win.cpp

```diff
@@ -315,11 +302,42 @@ DWORD WINAPI ListenerThread( LPVOID lpParam ) {
 	return 0;
 }
 
+BOOL GetDeviceDetails(HDEVINFO hDevInfo, SP_DEVINFO_DATA* pspDevInfoData, int index, PSP_DEVICE_INTERFACE_DETAIL_DATA* pspDeviceIntDetail) {
+	SP_DEVICE_INTERFACE_DATA spDevIntData;
+	spDevIntData.cbSize = sizeof(SP_DEVICE_INTERFACE_DATA);
+
+	if (!DllSetupDiEnumDeviceInterfaces(hDevInfo, NULL, &GUID_DEVINTERFACE_USB_DEVICE, index, &spDevIntData)) {
```

Comment:
Let's add a comment about what this check does. I am not sure but I am guessing if we can't enumerate the interface list then we can't get any details below.

Reply:
That is correct and I will add a comment to make that more clear.